### PR TITLE
MODINV-454 - IndexOutOfBoundsException observed on Data Import (Instances Tab)

### PR DESCRIPTION
**Pull request by RuslanLavrov**
## Purpose
to make the message on the json screen more informative and user-friendly when an action profile has not been linked to a mapping profile

## Approach
* add an additional check whether action profile has not been linked to mapping profile and provide user-friendly message for data-import log

## Learning
[MODINV-454](https://issues.folio.org/browse/MODINV-454)

### CODE REVIEW

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateHoldingEventHandler.java [hunk 1]
```diff
@@ -45,6 +45,7 @@
   private static final String SAVE_HOLDING_ERROR_MESSAGE = "Can`t save new holding";
   private static final String CONTEXT_EMPTY_ERROR_MESSAGE = "Can`t create Holding entity: context is empty or doesn`t exists";
   private static final String PAYLOAD_DATA_HAS_NO_INSTANCE_ID_ERROR_MSG = "Failed to extract instanceId from instance entity or parsed record";
+  static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to create holding entity has no a mapping profile";
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateHoldingEventHandler.java:48]:
```suggestion
  static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to create Holding entity requires a mapping profile";
```

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
 public class CreateInstanceEventHandler extends AbstractInstanceEventHandler {
 
   private static final String PAYLOAD_HAS_NO_DATA_MSG = "Failed to handle event payload - event payload context does not contain MARC_BIBLIOGRAPHIC data";
+  static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to create instance has no a mapping profile";
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java:36]:
```suggestion
  static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to create an Instance requires a mapping profile";
```

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java [hunk 3]
```diff
@@ -49,6 +49,7 @@
 
 public class CreateItemEventHandler implements EventHandler {
 
+  public static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to create an item has no a mapping profile";
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java:52]:
```suggestion
  public static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to create an Item requires a mapping profile";
```

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/ReplaceInstanceEventHandler.java [hunk 4]
```diff
@@ -35,6 +35,7 @@
 public class ReplaceInstanceEventHandler extends AbstractInstanceEventHandler { // NOSONAR
 
   private static final String PAYLOAD_HAS_NO_DATA_MSG = "Failed to handle event payload, cause event payload context does not contain MARC_BIBLIOGRAPHIC or INSTANCE data";
+  static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to update instance has no a mapping profile";
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/ReplaceInstanceEventHandler.java:38]:
```suggestion
  static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to update an Instance requires a mapping profile";
```

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateHoldingEventHandler.java [hunk 5]
```diff
@@ -37,6 +37,7 @@
   private static final String CONTEXT_EMPTY_ERROR_MESSAGE = "Can`t update Holding entity: context or Holding-entity are empty or doesn`t exist!";
   private static final String EMPTY_REQUIRED_FIELDS_ERROR_MESSAGE = "Can`t update Holding entity: one of required fields(hrid, permanentLocationId, instanceId) are empty!";
   private static final String HOLDINGS_PATH_FIELD = "holdings";
+  static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to update holding entity has no a mapping profile";
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateHoldingEventHandler.java:40]:
```suggestion
  static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to update a Holding entity requires a mapping profile";
```

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateItemEventHandler.java [hunk 6]
```diff
@@ -48,6 +48,7 @@
 
   private static final Logger LOG = LogManager.getLogger(UpdateItemEventHandler.class);
 
+  public static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to update an item has no a mapping profile";
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateItemEventHandler.java:51]:
```suggestion
  public static final String ACTION_HAS_NO_MAPPING_MSG = "Action profile to update an Item requires a mapping profile";
```

**Review by KaterynaSenchenko — APPROVED** [review]

**Comment by Aliaksandr-Fedasiuk** [comment]
Ruslan, could you, please, provide description to this PR?

**Review by Aliaksandr-Fedasiuk — APPROVED** [review]

**Pull request merged**